src/services/NodeMonitor.ts

```diff
@@ -124,49 +131,81 @@ export class NodeMonitor {
 
 			if (Array.isArray(nodePeers.data)) nodeList = [...nodePeers.data];
 		} catch (e) {
-			logger.error(`FetchNodePeersByURL. Failed to get /node/peers from "${hostUrl}". ${e.message}`);
+			logger.error(`[FetchNodePeersByURL] Failed to get /node/peers from "${hostUrl}". ${e.message}`);
 		}
 
 		return nodeList;
 	};
 
-	private getNodeListInfo = async () => {
-		logger.info(`Getting node from peers total ${this.nodeList.length} nodes`);
-		const nodeListChunks = splitArray(this.nodeList, this.nodeInfoChunks);
+	/**
+	 * Fetch peers from current node list and add them to the node list.
+	 */
+	private fetchAndAddNodeListPeers = async (): Promise<void> => {
+		const apiNodeList = this.nodeList.filter((node) => isAPIRole(node.roles));
 
-		this.nodeList = [];
+		logger.info(
+			`[fetchAndAddNodeListPeers] Getting peers from nodes, total nodes: ${this.nodeList.length}, api nodes: ${apiNodeList.length}`,
+		);
 
-		for (const nodes of nodeListChunks) {
-			logger.info(`Getting node info for chunk of ${nodes.length} nodes`);
+		await runTaskInChunks(this.nodeList, this.nodePeersChunkSize, logger, 'fetchAndAddNodeListPeers', async (nodes) => {
```

Comment:
It should use `apiNodeList` instead of `this.nodeList`?

Reply:
Interesting, I specifically created the `apiNodeList` for this use, must have been regressed during the refactoring, fixed now.